### ISSUE TITLE
MySql doesn't have a Last function

### ISSUE THREAD

**Issue by darozak** [Darwin-River/Ex-Machinis#147]
I need to figure out a way to work around the fact that MySql doesn't have a Last function.  How do I find the last increment/depletion event that relates to a particular drone and resource?

We ran into this problem before when developing the web query and Fernando was able to come up with a solution.

This discussion may also help: https://stackoverflow.com/questions/5495913/can-i-use-aggregation-function-last-in-mysql

**Comment by darozak** [comment]
Here's another helpful discussion thread: https://stackoverflow.com/questions/5688063/select-most-recent-row-with-group-by-in-mysql?noredirect=1&lq=1

**Comment by darozak** [comment]
After playing around with the query in Access and tweaking it to generate results which match the website, I came up with the following query:
```
SELECT o.resource, o.new_quantity
FROM (
  SELECT resource, MAX(timestamp) AS maxtime
  FROM events
  WHERE drone = [value_1] AND outcome = 1 AND (event_type =1 OR event_type = 2)
  GROUP BY drone, resource
) AS f
INNER JOIN events AS o ON o.timestamp = f.maxtime AND o.resource = f.resource
WHERE o.drone = [value_1]
ORDER BY o.new_quantity DESC;
```
This appears to work in all the situations that I tested.  

Here are some observations with respect to design:
* f stands for filter table and o stands for output table.
* The inner join needs to link both maxtime and resource.  This avoids issues where two different resource are dealt with at the same time.
* I needed to check for [value_1] in both f and o since at one point drones 52 and 53 exchanged resource 2 at the same time.  This lead to two entries in f with the same time.
* I also had to group by both drone and resource.
